fastestimator/estimator.py

```diff
@@ -62,14 +62,40 @@ def __init__(self,
         self.monitor_names = to_set(monitor_names)
         self.system = System(log_steps=log_steps, epochs=epochs, max_steps_per_epoch=max_steps_per_epoch)
         self.trace_inputs = dict()
+        self.traces_in_use = []
```

Comment:
why is this distinction necessary when we already have a system to inspect the mode of a trace?

Reply:
we need a variable to store the currently modified trace. (without overwriting the self.trace)

Reply:
why are the traces modified though?

Reply:
originally we have self.traces = self._prepare_traces()